atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresVersionCheck.java

```diff
@@ -15,34 +15,69 @@
  */
 package com.palantir.atlasdb.keyvalue.dbkvs.impl.postgres;
 
+import com.google.common.base.Suppliers;
+import com.palantir.logsafe.SafeArg;
 import com.palantir.util.AssertUtils;
 import com.palantir.util.VersionStrings;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.slf4j.Logger;
 
 public final class PostgresVersionCheck {
     static final String MIN_POSTGRES_VERSION = "9.6";
+    private static final Supplier<Pattern> VERSION_PATTERN = Suppliers.memoize(() -> {
+        // Matches: x, x.y, x.y.z, etc in group 1
+        // Allows discarding an optional additional section in parens, for example:
+        // 14.11 (Ubuntu 14.11-1.pgdg20.04+1)
+        return Pattern.compile("^([\\.0-9]+)( \\([^\\)]+\\))?$");
+    });
 
     private PostgresVersionCheck() {}
 
+    static Optional<String> extractValidPostgresVersion(String rawVersionString) {
```

Comment:
ah I see this is used in the test, we probably want the `@VisibleForTesting` annotation?